controllers/operatorcontext_controller_test.go

```diff
@@ -0,0 +1,143 @@
+package controllers_test
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/TykTechnologies/tyk-operator/api/model"
+	"github.com/TykTechnologies/tyk-operator/api/v1alpha1"
+	"github.com/TykTechnologies/tyk-operator/controllers"
+	"github.com/TykTechnologies/tyk-operator/pkg/keys"
+	"github.com/matryer/is"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestOperatorContextCreate(t *testing.T) {
+	is := is.New(t)
+
+	key := types.NamespacedName{
+		Name:      "test",
+		Namespace: "test-ns",
+	}
+	// Create a test object
+	opCtx := v1alpha1.OperatorContext{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test",
+			Namespace: "test-ns",
+		},
+	}
+
+	cl, err := controllers.NewFakeClient([]runtime.Object{&opCtx})
+	is.NoErr(err)
+
+	r := controllers.OperatorContextReconciler{
+		Client: cl,
+		Scheme: scheme.Scheme,
+		Log:    log.NullLogger{},
+	}
+
+	req := reconcile.Request{}
+	req.NamespacedName = key
+
+	_, err = r.Reconcile(context.TODO(), req)
+	is.NoErr(err)
+
+	var result v1alpha1.OperatorContext
+
+	err = cl.Get(context.TODO(), key, &result)
+	if err != nil {
+		t.Error(err)
+	}
+
+	is.True(len(result.Finalizers) != 0)
+	is.True(result.Finalizers[0] == keys.OperatorContextFinalizerName)
+}
+
+func TestOperatorContextDelete(t *testing.T) {
+	is := is.New(t)
+	dummyName := "dummy"
+
+	tests := map[string]struct {
+		Name        string
+		OpCtxStatus *v1alpha1.OperatorContextStatus
+		ReturnError bool
```

Comment:
Considering this is the initial work for unit tests, it looks good to me. But what do you think about checking for errors in `error` type instead of `bool` in the future? So, we can understand that we obtained the desired error instead of any errors.

Reply:
Updated tests.